runelite-client/src/main/java/net/runelite/client/plugins/driftnet/DriftNetPlugin.java

```diff
@@ -204,6 +221,30 @@ public void onGameTick(GameTick tick)
 			return;
 		}
 
+		for (DriftNet net : NETS)
```

Comment:
It feels like this entire thing can be reduced to a method that takes an npc which would let you utilize the `removeIf` below, and not have to create additional iterators.